### ISSUE TITLE
DNS lookup does not work when all traffic is blocked

### ISSUE THREAD

**Issue by grk-pancham** [rancher/rke2#2989]
DNS lookup for kubernetes.default works when all traffic rule is allowed in AWS NACL between rancher rke2 server master and worker nodes. however the DNS lookup fails when all traffic rule is removed and only the ports mentioned in the RKE2 network requirements document (https://docs.rke2.io/install/requirements/) are opened.  

Are there any other port that need to be opened to make DNS lookup work with core dns service in cluster? 

Kubernetes version: v1.23.1+rke2r1

kubectl exec -i -t dnsutils -- nslookup kubernetes.default
nslookup connection timed out no servers could be reached

**Comment by brandond** [comment]
Which CNI are you using? Did you open the ports specific to that CNI between nodes, and confirm that it's open between all nodes, not just between agents and servers?

**Comment by grk-pancham** [comment]
Yes. We use Calico CNI and have opened all the ports required by Calico as mentioned in the above doc.

Is Calico or CoreDns using any specific IP from our subnet CIDR (100.64.0.0) to communicate between the k8 nodes ?

**Comment by brandond** [comment]
If you're using Calico with VXLAN then everything going between the two nodes should be destined for UDP 4789.

I will note that in my experience using NACLs instead of Security Groups can be a bit fiddly, as you have to be really careful to explicitly allow return traffic. NACLs do not use session tracking as Security Groups do so you have to be sure that your rules allow the correct ports bidirectionally, without regard for which end initiated the traffic. This usually looks like allowing all traffic with a source OR destination of the port you need to allow. I think this is why most folks just use Security Groups instead of NACLs.

If you have VPC flow logs turned on, you should be able to see if you're dropping traffic between the nodes, and if so on what pords.
